dotcom-rendering/src/lib/article-count.ts

```diff
@@ -0,0 +1,32 @@
+import { storage } from '@guardian/libs';
+import {
+	getWeeklyArticleHistory,
+	incrementWeeklyArticleCount,
+} from '@guardian/support-dotcom-components';
+import { hasOptedOutOfArticleCount } from '../web/lib/contributions';
+import { incrementDailyArticleCount } from '../web/lib/dailyArticleCount';
+
+// Log an article view using the Slot Machine client lib
+// We should monitor this function call to ensure it only happens within an
+// article pages when other pages are supported by DCR.
+export const getArticleCount = async (pageId: string, keywordIds: string) => {
+	const hasOptedOut = await hasOptedOutOfArticleCount();
+
```

Comment:
We could just `return undefined` and stop execution. e.g.
```suggestion
if (hasOptedOut) { return }
```

which might make the below statement a bit nicer to understand i.e. update and set if not set.

Reply:
Or even:

```ts
if(await hasOptedOutOfArticleCount()) return;
```

Reply:
The existing solution still tries to get the article count even if the user is opted out, I'll have a look to see why and see if its possible to do this.

Reply:
The existing solution still tries to get the article count even if the user is opted out, I'll have a look to see why and see if its possible to do this.

Reply:
For context - the local storage item is deleted when a user opts out, so `getWeeklyArticleHistory` should return `undefined` in this case.
But this new function is a nice opportunity to be explicit about returning `undefined` if they've opted out